mlflow/tracing/fluent.py

```diff
@@ -164,28 +164,32 @@ def start_span(
     Returns:
         Yields an :py:class:`mlflow.tracing.MlflowSpanWrapper` that represents the created span.
     """
-    # TODO: refactor this logic
     try:
-        tracer = get_tracer(__name__)
-        span = tracer.start_span(name)
-    except Exception:
-        _logger.warning(f"Failed to start span with name {name}.")
-        span = None
+        otel_span = provider.start_span_in_context(name)
+
+        # If the trace is not registered in the in-memory trace manager, create a new trace.
+        trace_manager = InMemoryTraceManager.get_instance()
+        request_id = trace_manager.get_request_id_from_trace_id(otel_span.context.trace_id)
+        if not request_id:
+            trace_info = provider.create_trace_info(otel_span)
+            trace_manager.register_trace(otel_span.context.trace_id, trace_info)
+            request_id = trace_info.request_id
+
+        # Create a new MlflowSpanWrapper and register it to the in-memory trace manager.
+        mlflow_span = MlflowSpanWrapper(otel_span, request_id=request_id, span_type=span_type)
+        mlflow_span.set_attributes(attributes or {})
+        trace_manager.register_span(mlflow_span)
+
+    except Exception as e:
+        _logger.debug("Failed to start span: %s", e)
```

Comment:
```suggestion
        _logger.debug("Failed to start span: %s", e, , exc_info=True)
```

Reply:
If this was `_logger.warning`, I would not use `exc_info=True`.

Reply:
Makes sense!